src/Tokenizer/Tokens.php

```diff
@@ -516,19 +516,43 @@ public function getNextNonWhitespace(int $index, ?string $whitespaces = null): ?
     }
 
     /**
-     * Get index for closest next token of given kind.
+     * Get index for closest next token among given kinds.
      *
      * This method is shorthand for getTokenOfKindSibling method.
      *
-     * @param int $index token index
-     * @param list<array{int}|string|Token> $tokens        possible tokens
-     * @param bool $caseSensitive perform a case sensitive comparison
+     * @param int                                          $index         token index
+     * @param list<array{0: int, 1?: string}|string|Token> $tokens        possible tokens
+     * @param bool                                         $caseSensitive perform a case sensitive comparison
      */
     public function getNextTokenOfKind(int $index, array $tokens = [], bool $caseSensitive = true): ?int
     {
         return $this->getTokenOfKindSibling($index, 1, $tokens, $caseSensitive);
     }
 
+    /**
+     * Get index for closest next token of given kind.
+     *
+     * @param int                                        $index         token index
+     * @param array{0: int, 1?: string}|int|string|Token $tokens        possible tokens
+     * @param bool                                       $caseSensitive perform a case sensitive comparison
+     *
+     * @internal
+     */
+    public function tokenIndexOfKindAfter(int $index, $tokens, bool $caseSensitive = true): int
+    {
+        if (\is_int($tokens)) {
+            $tokens = [$tokens];
+        }
+
+        $nextIndex = $this->getNextTokenOfKind($index, [$tokens], $caseSensitive);
+
+        if (null === $nextIndex) {
+            throw new \LogicException("No token of given kind found after index {$index}.");
```

Comment:
I don't think it's a good exception type
ref https://www.php.net/manual/en/class.logicexception.php
> Exception that represents error in the program logic. This kind of exception should lead directly to a fix in your code.

We use those methods often in context "let's see if token of given type will be found, so we can fix it" - but if not, it's not logic issue of our code.

I'm not sure about exception for those methods in general. 
Let me compare it to `array_search` built-in feature, as it's very similar - in our case we search within internal array of tokens to find one that is matching given conditions. `array_search` returns empty results, not throwing exception, if item not found.

Reply:
That's the thing: these methods are supposed to be used in cases where we do know the requested token index exists. If it doesn't, that's an error in the program logic.

This removes the need to check if the result is `null`, as this result isn't expected at all.

Reply:
I think there could be some cases when it's actually optional, like here: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/ff49030e8508ea9b71091b6fdf94de858d898cdc/src/Tokenizer/Analyzer/WhitespacesAnalyzer.php#L27

Reply:
What would be optional?

Reply:
ping @keradus

Reply:
that there might not be a previous token of given type and it's totally fine in the example I shared - it's simply that there is no whitespace and its fine, and not that LogicException should be raised and someone should fix the code

Reply:
Yes, this example is a scenario where not findind the expected token would be normal so using this new method would be incorrect.

Reply:
@keradus Are you OK with the current state of this PR or do you want me to make some changes?

Reply:
@julienfalque , overall, I see huge value in ensuring even more strict typing and less chances for unexpected crash, as right now it's easy to forget checking if index was actually found. Same time, it's not that handy to always checking `!== null` afterwards.

Yet, I'm not sure of the direction taken here, especially as right now the added code is not in use. While I believe this PR found good places to improve, I don't want to rush with the conclusion myself (yeah, i know this PR is not 1 day old, but also I cannot commit few hours to think about it, as I would like to).

IMHO, we should understand and align on the following - I let myself to split into threads, let's discuss there:
- exception - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6680/files#r1057991606
- single vs plural items - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6680/files#r1057992546
- future expected interface - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6680/files#r1057992920